react/week4/searcher/src/SearchResult.js

```diff
@@ -0,0 +1,37 @@
+import {useContext} from 'react';
+import {SearchContext} from './SearchContext';
+
+const SearchResult = () => {
+    const searchContext = useContext(SearchContext);
+
+    return (
+        <div>
+            {searchContext.isLoading
+            ? 'Loading...'
+            : searchContext.error.length > 0
+                ? <p>{searchContext.error}</p>
+                : <UserList />
+            }
+        </div>
+    )
+}
+
+const UserList = () => {
+    const searchContext = useContext(SearchContext);
+    const {users} = searchContext;
+
+    return (
+        <div>
+            {users === undefined || users.length === 0
+                ? 'No users'
+                : <ul>
+                    {users.map(user => {
+                        return <li key={user.id}>{user.login}</li>
+                    })}
+                </ul>
+            }
```

Comment:
There's a "cheat" you can maybe use called [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).
It allows you to check for falsy values in the call like this:
`users?.length === 0`
This will check if users is defined before is checks the length.

Reply:
Thank you so much for your feedback! I handled your comments and made changes to my app, but unfortunately, this optional chaining cheat doesn`t work.